BraceExpansion.java

```diff
@@ -0,0 +1,156 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+// "static void main" must be defined in a public class.
+//Method 1 , Sorting the result at last
+public class Main {
+
+    // Time Complexity : 0(k^n);n is the no. of blocks and k is the average length.
+// Space Complexity : 0(nk)
+// Did this code successfully run on Leetcode : No, I don't have premium
+// Any problem you faced while coding this : No
+
```

Comment:
Solution is correct

Reply:
Thank you for your time and going through the solution. Appreciate it.

Also if possible, we had a disagreement on TC and SC on some of the PR's I had submitted a month back. I had written down my thought process for getting the TC and SC. If possible could you please go over those and let me know if I am correct or not?